packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+        uint256 jurorFees = _amount.sub(governorFees);
+        period.jurorFees[_token] = period.jurorFees[_token].add(jurorFees);
+
+        // Deposit tokens from sender to this contract
+        _deposit(msg.sender, _token, _amount);
+        emit PaymentReceived(periodId, _payer, _token, _amount, msg.sender, _data);
+    }
+
+    /**
+    * @notice Claim jurors fees for period #`_periodId` owed to `msg.sender`
+    * @param _periodId Identification number of the period which fees are claimed for
+    * @param _token Address of the token to be claimed
+    */
+    function claimJurorFees(uint256 _periodId, address _token) external {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        require(!_hasClaimedJurorFees(period, msg.sender, _token), ERROR_JUROR_FEES_ALREADY_CLAIMED);
+
+        (uint64 periodBalanceCheckpoint, uint256 totalActiveBalance) = _ensurePeriodBalanceDetails(period, _periodId);
+        uint256 jurorActiveBalance = _getJurorActiveBalance(msg.sender, periodBalanceCheckpoint);
+        uint256 amount = _getJurorFees(period, _token, jurorActiveBalance, totalActiveBalance);
+        _claimJurorFees(period, _periodId, msg.sender, _token, amount);
+    }
+
+    /**
+    * @notice Claim juror fees for period #`_periodId` owed to `msg.sender`
+    * @dev It will ignore tokens that were already claimed without reverting
+    * @param _periodId Identification number of the period which fees are claimed for
+    * @param _tokens List of token addresses to be claimed
+    */
+    function claimManyJurorFees(uint256 _periodId, address[] calldata _tokens) external {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        (uint64 periodBalanceCheckpoint, uint256 totalActiveBalance) = _ensurePeriodBalanceDetails(period, _periodId);
+        uint256 jurorActiveBalance = _getJurorActiveBalance(msg.sender, periodBalanceCheckpoint);
+
+        // We assume the token contract is not malicious
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            address token = _tokens[i];
+            if (!_hasClaimedJurorFees(period, msg.sender, token)) {
+                uint256 amount = _getJurorFees(period, token, jurorActiveBalance, totalActiveBalance);
+                _claimJurorFees(period, _periodId, msg.sender, token, amount);
+            }
+        }
+    }
+
+    /**
+    * @notice Transfer owed fees to the governor for period #`_periodId`
+    * @param _periodId Identification number of the period being claimed
+    * @param _token Address of the token to be claimed
+    */
+    function transferGovernorFees(uint256 _periodId, address _token) external {
+        require(_periodId <= _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        address payable governor = address(uint160(_configGovernor()));
```

Comment:
Doesn't `_configGovernor()` already return an address?

Reply:
Yes but it is not payable, am I missing sth?

Reply:
Nevermind me, just realizing now that in 0.5 we have to cast back to an `uint160` first!